### PR TITLE
docs(ifc): add edge case

### LINKED ISSUE #628: `MorphoBalancesLib.expectedBorrowAssets` may return a value greater than `totalBorrowAssets`
```solidity
    function testExpectedBorrowAssetsLeTotalBorrowAssets(uint256 borrowedShares) public {
        borrowedShares = bound(borrowedShares, 1, SharesMathLib.VIRTUAL_SHARES - 1);

        collateralToken.setBalance(BORROWER, 1);

        oracle.setPrice(1e37);

        vm.startPrank(BORROWER);
        morpho.supplyCollateral(marketParams, 1, BORROWER, hex"");
        morpho.borrow(marketParams, 0, borrowedShares, BORROWER, BORROWER);
        vm.stopPrank();

        (,, uint256 totalBorrowAssets,) = morpho.expectedMarketBalances(marketParams);

        assertLe(morpho.expectedBorrowAssets(marketParams, BORROWER), totalBorrowAssets);
    }
```

Run: `yarn test:forge -vvv --mt testExpectedBorrowAssetsLeTotalBorrowAssets`

This is due to the `toAssetsUp`

Here is another test that shows that this happens whatever `totalBorrowAssets` (not only where it is zero):

```solidity
    function testAssetsUp(uint256 shares, uint256 totalAssets, uint256 totalShares) public {
        totalAssets = bound(totalAssets, 0, type(uint128).max);
        totalShares = bound(totalShares, 1, type(uint128).max);
        shares = bound(shares, 1, totalShares);

        vm.assume(totalAssets >= 1 ether);

        assertLe(shares.toAssetsUp(totalAssets, totalShares), totalAssets);
    }
```

The end stakes are:
1. Underflow upon repay/liquidate because `totalBorrowAssets < repaidAssets`, when using the return value as input
2. This rounding seems to only happen when `shares ~= totalShares +- 10`

(1) is addressed via https://github.com/morpho-labs/morpho-blue-private/blob/f463e40f776acd0f26d0d380b51cfd02949c8c23/src/libraries/periphery/MorphoBalancesLib.sol#L108-L109

(2) is unlikely because it requires the borrow shares to be concentrated to a single address

### PR DESCRIPTION
**Pull request by Rubilmax**
- Fixes #628

### CODE REVIEW

**Review by MerlinEgalite — APPROVED** [review]

File: src/libraries/periphery/MorphoBalancesLib.sol [hunk 1]
```diff
@@ -105,8 +105,10 @@ library MorphoBalancesLib {
     }
 
     /// @notice Returns the expected borrow assets balance of `user` on a market after having accrued interest.
-    /// @dev Warning: repaying a borrow position using the expected assets balance can lead to a revert due to
-    /// conversion roundings between shares and assets.
+    /// @dev Warning: The expected balance is rounded up, so it can be greater than the market's expected total supply
```
Comment (QGarchery) [inline src/libraries/periphery/MorphoBalancesLib.sol:108]:
```suggestion
    /// @dev Warning: The expected balance is rounded up, so it can be greater than the market's expected total borrow
```

File: src/libraries/periphery/MorphoBalancesLib.sol [hunk 2]
```diff
@@ -105,8 +105,10 @@ library MorphoBalancesLib {
...
+    /// assets.
+    /// @dev Warning: Repaying using the expected supply assets can lead to a revert due to conversion roundings from
+    /// assets to shares.
```
Comment (QGarchery) [inline src/libraries/periphery/MorphoBalancesLib.sol:111]: Do we have a concrete example of that ? It may not be an issue because of the zeroFloorSub/min in repay/liquidate

Comment (Rubilmax) [reply src/libraries/periphery/MorphoBalancesLib.sol]: Oh, that's right, so maybe this is what the comment was about, but badly phrased

**Review by MerlinEgalite — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Review by QGarchery — APPROVED** [review]

**Review by MathisGD — APPROVED** [review]

**Comment by MathisGD** [comment]
@Jean-Grimal can you review please

**Review by Jean-Grimal — APPROVED** [review]
